slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/dao/ValidatorsDao.java

```diff
@@ -64,4 +65,26 @@ public boolean isEnabled(final Handle handle, final int validatorId) {
         .mapTo(Boolean.class)
         .first();
   }
+
+  public boolean hasSigned(final Handle handle, final Bytes publicKey) {
+
+    Optional<Integer> maybeValidatorId =
```

Comment:
This shouldn't be necessary to look up the validatorId as we have the validator id in the slashing protection service and can pass that in

Reply:
done 👍